src/Nethermind/Nethermind.Runner/Program.cs

```diff
@@ -591,4 +565,69 @@ private static string GetProductInfo()
 
         return info.ToString();
     }
+
+    private static bool ValidateArguments(string[] args, CommandLineApplication app)
+    {
+        static HashSet<string> GetValidArguments(CommandLineApplication app)
+        {
+            var validArguments = new HashSet<string>();
+            foreach (var option in app.GetOptions())
+            {
+                if (!string.IsNullOrEmpty(option.LongName))
+                {
+                    validArguments.Add(option.LongName);
+                }
+                if (!string.IsNullOrEmpty(option.ShortName))
+                {
+                    validArguments.Add(option.ShortName);
+                }
+            }
+
+            return validArguments;
+        }
+
+        static ReadOnlyMemory<char> GetArgumentName(string arg) => arg.StartsWith("--") ? arg.AsMemory(2) : arg.StartsWith('-') ? arg.AsMemory(1) : ReadOnlyMemory<char>.Empty;
+        static IEnumerable<ReadOnlyMemory<char>> GetArgumentNames(IEnumerable<string> args)
+        {
+            bool lastWasArgument = false;
+            foreach (ReadOnlyMemory<char> potentialArgument in args.Select(GetArgumentName))
+            {
+                if (!lastWasArgument)
+                {
+                    bool isCurrentArgument = lastWasArgument = !potentialArgument.IsEmpty;
+                    if (isCurrentArgument)
+                    {
+                        yield return potentialArgument;
+                    }
+                }
+                else
+                {
+                    lastWasArgument = false;
+                }
+            }
+        }
+
+        // Get all valid options from the configuration files
+        var validArguments = GetValidArguments(app);
+
+        var argumentsNamesProvided = GetArgumentNames(args).ToList();
```

Comment:
Try not to use `var` unless the types are anonymous or very long and obvious, makes harder to read and understand what is going on in GitHub

Reply:
Not quite sure I understand
Do you mean avoid using var and try to set an explicit type when possible?

The last part `makes harder to read and understand what is going on in GitHub` it's the part that confuses me

Reply:
@wugalde19 on Github, you can't infer the type or jump into the function definition to see what it is. 

It makes sense to user var if it is evident, like GetString(), that it should return some string. 

For the example in question, what does return `GetValidArguments`? The context of the changes does not make it clear.  

Makes more sense? I think it's what @LukaszRozmej was referring to.

Reply:
Ohhhh I see
Of course, the idea is to provide more context of what a variable will contain unless it's easy to infer.
Yeah yeah makes sense
And that part about Github it's like that you cannot jump on Github between definitions easily as you can do it in some IDEs which makes things harder in terms of knowing/checking what a method returns and all of that

Perfect
Thanks for clarifying this to me @stdevMac